vowpalwabbit/core/include/vw/core/automl_impl.h

```diff
@@ -217,6 +218,7 @@ class interaction_config_manager
   bool reward_as_cost;
   double tol_x;
   bool is_brentq;
+  VW::interactions_generator _interactions_cache;
```

Comment:
this is public, should it be without the underscore in the name or made private?

Reply:
automl is in a weird state where we need a lot of things to be public just to facilitate asserting on internal state on the unit tests, to try to keep control of edge cases and specific states we now should be true after X events.

I haven't figured out a better way to allow the unit test to have full access.